doc/datasets/rcv1.rst

```diff
@@ -32,7 +32,7 @@ The target values are stored in a scipy CSR sparse matrix, with 804414 samples a
 Each sample can be identified by its ID, ranging (with gaps) from 2286 to 810596::
 
     >>> rcv1.sample_id[:3]
-    array([2286, 2287, 2288], dtype=int32)
```

Comment:
Good point. I am digging into this, but I must be missing something. I created a conda environment with the same come configuration of this travis test [here](https://github.com/scikit-learn/scikit-learn/blob/master/.travis.yml#L43-L45) which still breaks on my machine:
```
Failed example:
    rcv1.sample_id[:3]
Expected:
    array([2286, 2287, 2288], dtype=int32)
Got:
    array([2286, 2287, 2288], dtype=uint32)
```